FRAuthenticator/FRAuthenticator/Model/Notification/PushNotification.swift

```diff
@@ -290,12 +378,15 @@ public class PushNotification: NSObject, NSSecureCoding, Codable {
     }
     
     
-    func buildPushAuthenticationRequest(approved: Bool, mechanism: PushMechanism) throws -> Request {
+    func buildPushAuthenticationRequest(challengeResponse: String? = nil, approved: Bool, mechanism: PushMechanism) throws -> Request {
         var payload: [String: CodableValue] = [:]
         payload[FRAConstants.response] = try CodableValue(Crypto.generatePushChallengeResponse(challenge: self.challenge, secret: mechanism.secret))
-        if !approved {
+        if self.pushType == .default && !approved {
```

Comment:
The deny in the payload will be included for all Push types if the notification is rejected. So, I think this change is not required.

Reply:
Sounds good. Will revert it.